src/lib/response-triggers/userlocation.ts

```diff
@@ -0,0 +1,334 @@
+				return 'Error setting location';
+			}
+		}
+
+		if (users.some(x => !x.isMention)) {
+			return 'Mixed mentions and non mentions not allowed';
+		}
+
+		const usersPromise = users.map(async x => {
+			try {
+				await this.setUserLocation(x.userId, location);
+				return {
+					user: x.text,
+					success: true
+				};
+			} catch (error) {
+				Logger.error(`[UserLocation.setUserLocation] Error setting location for user ${x.userId}: ${error.message}`);
+				return {
+					user: x.text,
+					success: false
+				}
+			}
+		});
+
+		const result = await Promise.all(usersPromise);
+		if (result.some(x => !x.success)) {
+			return `Location setting was unsuccessful for user(s): ${result.filter(x => !x.success).map(x => `'${x.user}'`).join(', ')}`;
+		}
+
+		return 'Location has been set';
+	}
+
+	private async setUserLocation(userId : string, location : string) : Promise<void> {
+		try {
+			let exists = await this.dbUsers.existsOrCanBeCreated(userId);
+			if (!exists) {
+				throw new Error(`User ${userId} could not be found or created.`);
+			}
+		} catch (error) {
+			Logger.error(`[UserLocation.setUserLocation] Error finding or creating user ${userId}: ${error.message}`);
+			throw new Error(`Error: User ${userId} could not be found or created.`);
+		}
+
+		await this.dbLocation.setUserLocation(userId, location);
+	}
+
+	private async deleteUserLocationAdmin(args : Argument[], message : MessageObject) : Promise<string> {
+		if (args.length < 4) {
+			return 'Please specify a list of mentions/me';
+		}
+
+		const users = args.filter((item, index) => index > 2 );
+
+		if (users.length === 1 && !users[0].isMention) {
+			if (users[0].text !== 'me') {
+				return `Invalid person ${users[0].text}. Either specify 'me' or mention a person`;
+			}
+
+			try {
+				await this.dbLocation.deleteUserLocation(message.personId);
+				return 'Location has been deleted';
+			} catch (error) {
+				Logger.error(`[UserLocation.setUserLocation] Error deleting location for user ${message.personId}: ${error.message}`);
+				return 'Error deleting location';
+			}
+		}
+
+		if (users.some(x => !x.isMention)) {
+			return 'Mixed mentions and non mentions not allowed';
+		}
+
+		const usersPromise = users.map(async x => {
+			try {
+				await this.dbLocation.deleteUserLocation(x.userId);
+				return {
+					user: x.text,
+					success: true
+				};
+			} catch (error) {
+				Logger.error(`[UserLocation.setUserLocation] Error deleting location for user ${x.userId}: ${error.message}`);
+				return {
+					user: x.text,
+					success: false
+				}
+			}
+		});
+
+		const result = await Promise.all(usersPromise);
+		if (result.some(x => !x.success)) {
+			return `Location deleting was unsuccessful for user(s): ${result.filter(x => !x.success).map(x => `'${x.user}'`).join(', ')}`;
+		}
+
+		return 'Location has been deleted';
+	}
+
+	private async getAllUserLocationMessage() : Promise<string> {
```

Comment:
Given the size of this file, it might have been better to chuck some things in a userLocationService - but I'm fine with this if it's too much trouble now.